src/OpenTelemetry/Logs/LogRecord.cs

```diff
@@ -213,22 +221,77 @@ public void ForEachScope<TState>(Action<LogRecordScope, TState> callback, TState
             return ref this.Data;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal void ResetReferenceCount()
+        {
+            this.PoolReferenceCount = 1;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal void AddReference()
+        {
+            Interlocked.Increment(ref this.PoolReferenceCount);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal int RemoveReference()
+        {
+            return Interlocked.Decrement(ref this.PoolReferenceCount);
+        }
+
+        internal void Buffer()
+        {
+            this.BufferLogStateValues();
```

Comment:
add a comment/todo for "state" - do we need to do similar for state as well?

Reply:
Updated